lib/pushmi_pullyu/swift_depositer.rb

```diff
@@ -23,27 +23,30 @@ def deposit_file(file_name, swift_container)
 
     era_container = swift_connection.container(swift_container)
 
-    deposited_file = if era_container.object_exists?(file_base_name)
-                       era_container.object(file_base_name)
-                     else
-                       era_container.create_object(file_base_name)
-                     end
+    object_exists = era_container.object_exists?(file_base_name)
 
     # Add swift metadata with in accordance to AIP spec:
     # https://docs.google.com/document/d/154BqhDPAdGW-I9enrqLpBYbhkF9exX9lV3kMaijuwPg/edit#
-    metadata = {
+    swift_metadata = {
       project: 'ERA',
       project_id: file_base_name,
       promise: 'bronze',
       aip_version: '1.0'
     }
 
     # ruby-openstack wants all keys of the metadata to be named like "X-Object-Meta-{{Key}}", so update them
-    metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
+    swift_metadata.transform_keys! { |key| "X-Object-Meta-#{key}" }
 
-    deposited_file.write(File.open(file_name),
-                         { 'etag' => checksum,
-                           'content-type' => 'application/x-tar' }.merge(metadata))
+    # creatre file metadata
+    file_metadata = { 'etag' => checksum,
+                      'content-type' => 'application/x-tar' }.merge(swift_metadata)
+
+    # create new file
+    return era_container.create_object(file_base_name, file_metadata, File.open(file_name)) unless object_exists
```

Comment:
Using return makes this hard to follow as you are abrupting the code flow and returning early. Also if someone comes into this file and wonders what this returns and sees the deposited_file variable being returned at the bottom, they will make the wrong assumption that this is what gets returned on a new file upload. This could all be simplified as such:

```ruby

    if era_container.object_exists?(file_base_name)
      deposited_file = era_container.object(file_base_name)
      deposited_file.write(File.open(file_name), file_metadata)
    else
      deposited_file = era_container.create_object(file_base_name, file_metadata, File.open(file_name))
    end

    deposited_file
```
Therefore you are always returning the same thing and the code flows much better (instead of being interrupted early and returning before it hits the end block)